source/manual/logging.html.md

```diff
@@ -147,8 +147,51 @@ AND "request_received"
 GROUP BY "status";
 ```
 
+### Adding a new field to the CDN logs
+
+Adding a new field to the CDN logs is a half manual, half automated
+process and is [tracked as tech debt](https://trello.com/c/7pAvfM8R/167).
+
+You should do this in the Staging environment first and wait until you see the
+logs coming through to Athena from Fastly. Only then should you change
+Production, to avoid syntax errors and incidents.
+
+Firstly, make the manual changes via Terraform and a PR. This ensures that you've
+had two pairs of eyes on both the Athena/AWS Glue config changes and the eventual JSON
+you will manually put into the Fastly web UI:
+
+1. Edit the [`infra-fastly-logs` Terraform](https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/infra-fastly-logs/main.tf).
+1. Find the `aws_glue_catalog_table` resource for the Fastly logs you want to add a column to (`govuk_www`, `govuk_assets` or `govuk_bouncer`).
+1. Copy the VCL for the log from [Fastly's list of available logs][fastly-logs-list] and add it to the list of commented (`//`) columns. It's important that these are in the same format as the preceeding commented lines (that is, JSON, with quotes) otherwise Athena won't parse the logs correctly. Here's an [example for the `govuk_www.cache_response` column](https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/infra-fastly-logs/main.tf#L251).
+1. Add your new column name, type and description to the JSON below the comments ([example for the `govuk_www.cache_response` column](https://github.com/alphagov/govuk-aws/blob/master/terraform/projects/infra-fastly-logs/main.tf#L330-L334).
+1. Raise a PR, get a review, merge and [deploy](manual/deploying-terraform.html).
+
+The manual steps to make Fastly send the log data:
+
+1. Log in to Fastly using the credentials from `fastly/deployment_shared_credentials` in the 2ndline password store.
+1. Search for the environment, for example "staging".
+1. On the environment page, click "Real Time Stats".
+1. Click "Configure" on the page with the graphs.
+1. Click the blue "Edit Configuration" button on the far right to reveal a dropdown menu.
+1. Click "Clone version xxx (active) to edit".
+1. Click "Logging" in the left hand menu.
+1. Find and click the link for the relevant log, for example "GOV.UK Fastly Logs S3 Bucket".
+1. Paste the new log format (found [in the list of available Fastly logs][fastly-logs-list]) into the "Log Format" text box.
+1. Click "Update".
+1. Click the purple "ACTIVATE" button in the top right corner to make the new configuration live.
+
+Both these sets of steps must be done! Check the S3 bucket and
+query Athena to see the added column and confirm that there's data for
+it. It can take up to half an hour for Athena to recognise that there have
```

Comment:
I think you may have got this half an hour from me from something different I was referring too. It can take up to half an hour for Fastly to send us through a log file with the changed data (we've set it to send them to us every 15 minutes but they seem to often come 15 minutes after the time they stopped logging).

For Athena to recognise the change in table that could potentially take up to 4 hours as that is how often the crawler worker runs. I don't know off the top of my head what happens though at the point of you running the terraform to add the new column. It might be that it is instantaneous and finds that in any data that has it or that it has to wait for the crawler to run which would be within 4 hours.

Sorry for vagueness I've not actually added a new column myself to know what problems it may cause. Simplest thing may be to just suggest that someone runs the crawler worker once they know Fastly is sending the right data.

Reply:
Done! Thanks for the good explanation!